ReactAndroid/src/main/java/com/facebook/react/uimanager/BaseViewManager.java

```diff
@@ -84,6 +86,12 @@ public void setElevation(T view, float elevation) {
     // Do nothing on API < 21
   }
 
+  @ReactProp(name = PROP_Z_INDEX)
+  public void setZIndex(T view, float zIndex) {
```

Comment:
Docs for `@ReactProp` says it can be a float but yeah, just test it and if it doesn't work don't worry about it and use float.